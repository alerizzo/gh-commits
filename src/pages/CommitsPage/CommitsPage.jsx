import React, { useEffect, useState } from 'react';
import { fetchCommits } from 'lib/GitHubAPI/Commits';

import CommitsPageHeader from './components/CommitsPageHeader';
import { CommitsList } from 'components';

function CommitsPage({ repository }) {
  const [commits, setCommits] = useState(null);
  const [error, setError] = useState(null);
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    if (repository && (!commits || commits.url !== repository.url)) {
      setLoading(true);
      fetchCommits(repository)
        .then(result => {
          // clear previous error, if any
          if (error) setError(null);

          // store result in state, and add repository metadata
          setCommits({ ...repository, ...result });
          setLoading(false);
        })
        .catch(err => {
          setError(err);
          setLoading(false);
        });
    }
  }, [repository, commits, error]);

  const handleSearch = () => {};

  const isBlank = !loading && !repository,
    isEmpty = !loading && !!repository && commits && commits.entries.length === 0,
    isError = !loading && !!error,
    isOk = !loading && !error && commits && commits.entries.length > 0;

  console.dir(commits);

  return (
    <div className="CommitsPage">
      <CommitsPageHeader
        onSearch={handleSearch}
        repository={repository}
        disabled={!isOk}
        showTableHeader={isOk}
      />
      {isOk && <CommitsList commits={commits.entries} />}
    </div>
  );
}

export default CommitsPage;
