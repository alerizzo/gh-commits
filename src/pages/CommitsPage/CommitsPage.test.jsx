import React from 'react';
import renderer from 'react-test-renderer';

import CommitsPage from './CommitsPage';

describe('<CommitsPage />', () => {
  const defaultProps = {};
  const wrapper = renderer.create(<CommitsPage {...defaultProps} />);

  test('render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
