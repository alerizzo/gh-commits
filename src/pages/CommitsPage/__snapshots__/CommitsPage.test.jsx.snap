// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommitsPage /> render 1`] = `
<div
  className="Page CommitsPage"
>
  <div
    className="CommitsPageHeader"
  >
    <h1
      className="title is-3"
    >
      Commits List
    </h1>
    <div
      className="field"
    >
      <label
        className="label is-size-6"
      >
        Search
      </label>
      <div
        className="TextInput control "
        data-error={null}
        style={
          Object {
            "width": "276px",
          }
        }
      >
        <input
          className="input  "
          disabled={true}
          onChange={[Function]}
          placeholder="Search ..."
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="CommitsList_Placeholder"
  >
    <div
      className="CommitsList"
    >
      <div
        className="columns CommitRow"
      >
        <div
          className="column is-2"
        >
          <div
            className="Skeleton Skeleton_Circle "
            style={
              Object {
                "height": "28px",
                "width": "28px",
              }
            }
          />
          <div
            className="Skeleton Skeleton_Text "
            style={
              Object {
                "width": "60%",
              }
            }
          />
        </div>
        <div
          className="column is-2"
        >
          <div
            className="Skeleton Skeleton_Text "
            style={
              Object {
                "width": "50%",
              }
            }
          />
        </div>
        <div
          className="column is-5"
        >
          <div
            className="Skeleton Skeleton_Text "
            style={
              Object {
                "width": "70%",
              }
            }
          />
        </div>
        <div
          className="column is-3"
        >
          <div
            className="Skeleton Skeleton_Text "
            style={
              Object {
                "width": "50%",
              }
            }
          />
        </div>
      </div>
      <div
        className="columns CommitRow"
      >
        <div
          className="column is-2"
        >
          <div
            className="Skeleton Skeleton_Circle "
            style={
              Object {
                "height": "28px",
                "width": "28px",
              }
            }
          />
          <div
            className="Skeleton Skeleton_Text "
            style={
              Object {
                "width": "60%",
              }
            }
          />
        </div>
        <div
          className="column is-2"
        >
          <div
            className="Skeleton Skeleton_Text "
            style={
              Object {
                "width": "50%",
              }
            }
          />
        </div>
        <div
          className="column is-5"
        >
          <div
            className="Skeleton Skeleton_Text "
            style={
              Object {
                "width": "70%",
              }
            }
          />
        </div>
        <div
          className="column is-3"
        >
          <div
            className="Skeleton Skeleton_Text "
            style={
              Object {
                "width": "50%",
              }
            }
          />
        </div>
      </div>
      <div
        className="columns CommitRow"
      >
        <div
          className="column is-2"
        >
          <div
            className="Skeleton Skeleton_Circle "
            style={
              Object {
                "height": "28px",
                "width": "28px",
              }
            }
          />
          <div
            className="Skeleton Skeleton_Text "
            style={
              Object {
                "width": "60%",
              }
            }
          />
        </div>
        <div
          className="column is-2"
        >
          <div
            className="Skeleton Skeleton_Text "
            style={
              Object {
                "width": "50%",
              }
            }
          />
        </div>
        <div
          className="column is-5"
        >
          <div
            className="Skeleton Skeleton_Text "
            style={
              Object {
                "width": "70%",
              }
            }
          />
        </div>
        <div
          className="column is-3"
        >
          <div
            className="Skeleton Skeleton_Text "
            style={
              Object {
                "width": "50%",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="overlay is-size-5 has-text-weight-bold"
    >
      Your commits will show up here
    </div>
  </div>
</div>
`;
